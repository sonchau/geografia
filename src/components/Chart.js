import React from 'react';
import ReactDOM from 'react-dom';
import {connect} from 'react-redux';
import {Bar, Line, HorizontalBar} from 'react-chartjs-2';
import {getData} from '../selectors/data';

export class Chart extends React.Component {

    render() {
        const chartData = {
            data: {
                labels: this.props.data.label,
                datasets: [
                {
                    label: 'Male',
                    backgroundColor: 'blue',
                    hoverBackgroundColor: 'rgba(4,12,96,0.4)',
                    data: this.props.data.male
                },
                {
                    label: 'Female',
                    backgroundColor: 'red',
                    hoverBackgroundColor: 'rgba(94,2,2,0.4)',
                    data: this.props.data.female

                }
                ]
            }
        }
        const options = {
            title: {
                display: true,
                text: 'Age Sex Chart',
                fontSize: 40,
            },
            legend: {
                display: true,
                position: 'top'
            }
        }

        return (
            <div className="chart">
            <Bar data={chartData.data} options={options}/>
          </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
      data : getData(state.data)
    }
}

export default connect(mapStateToProps)(Chart);
