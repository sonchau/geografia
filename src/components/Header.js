import React from 'react';

export class Header extends React.Component {

  render () {
    return (
      <header className="header">
      <div className="content-container">
        <div className="header__content">
          <span className="header__message">Welcome </span> 
        </div>
      </div>
    </header>
    )
  }
}

