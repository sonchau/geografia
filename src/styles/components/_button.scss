.button {
  background: $blue;
  border: none;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
  margin: $s-size;
  cursor: pointer;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888;
}

button:disabled {
  cursor: default;
  background: #6c757d;
}
