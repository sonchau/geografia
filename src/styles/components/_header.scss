.header {
  background: $dark-blue;
}

.header__content {
  text-align: center;
  padding: $s-size 0;
}

.header__message {
  color: white;
  font-size: $l-size;
}