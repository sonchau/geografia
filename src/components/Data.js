import React from 'react';
import {connect} from 'react-redux';
import {startSetData} from '../actions/data';
import Chart from './Chart';

export class Data extends React.Component {

    handleOnClick = (props) => {
        this.props.handleOnClick(props.target.value);      
    }

    render(props) {
        return (
            <div className="content-container">
                <Chart />
                <button className="button" value="dlgc_2011_ste_demo_age5gend" onClick={this.handleOnClick}> Year 2011</button>
                <button className="button" value="dlgc_2016_ste_demo_age5gend" onClick={this.handleOnClick}> Year 2016</button>
            </div>    
        )                  
    }
}
const mapDispatchToProps = (dispatch) => ({
    handleOnClick: (year) => dispatch(startSetData(year))
  });
export default connect(undefined, mapDispatchToProps)(Data);