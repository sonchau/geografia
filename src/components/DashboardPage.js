import React from 'react';
import {Header} from '../components/Header';
import Data from '../components/Data';

const DashboardPage = () => (
  <div>
    <Header />
    <Data/>
  </div>
);

export default DashboardPage;
