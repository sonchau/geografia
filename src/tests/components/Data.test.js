import React from 'react';
import { shallow } from 'enzyme';
import { Data } from '../../components/Data';

test('should render Data correctly', () => {
  const wrapper = shallow(<Data />);
  expect(wrapper).toMatchSnapshot();
});

test('should handle click button correctly', () => {
  
  const handleOnClick = jest.fn();
  const value = 'dlgc_2011_ste_demo_age5gend';
  const wrapper = shallow(<Data handleOnClick={handleOnClick} />);
  wrapper.find('button').at(0).simulate('click', {
    target: { value }
  });
  expect(handleOnClick).toHaveBeenCalled();
});