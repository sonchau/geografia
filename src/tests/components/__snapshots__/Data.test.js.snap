// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Data correctly 1`] = `
<div
  className="content-container"
>
  <Connect(Chart) />
  <button
    className="button"
    onClick={[Function]}
    value="dlgc_2011_ste_demo_age5gend"
  >
     Year 2011
  </button>
  <button
    className="button"
    onClick={[Function]}
    value="dlgc_2016_ste_demo_age5gend"
  >
     Year 2016
  </button>
</div>
`;
